src/cargo/ops/cargo_read_manifest.rs

```diff
@@ -112,7 +112,13 @@ fn read_nested_packages(path: &Path,
     let manifest = try!(find_project_manifest_exact(path, "Cargo.toml"));
 
     let (pkg, nested) = try!(read_package(&manifest, source_id, config));
-    all_packages.insert(pkg.package_id().clone(), pkg);
+    let pkg_id = pkg.package_id().clone();
+    if !all_packages.contains_key(&pkg_id) {
+        all_packages.insert(pkg_id, pkg);
+    } else {
+        warn!("skipping nested package `{}` found at `{}`",
+              &pkg.name(), &path.to_string_lossy());
```

Comment:
You should be able to omit the leading `&` characters here.

Can you also switch this to a `info!` from `warn!`?